internal/plugin/resolver.go

```diff
@@ -81,30 +81,70 @@ func (resolver *Resolver) Resolve(credentials []*config_v2.Credential) (map[stri
 	errorStrings := make([]string, 0, len(credentials))
 
 	var err error
+
+	// Group credentials by provider
+	var credentialsByProvider = make(map[string][]*config_v2.Credential)
 	for _, credential := range credentials {
-		var provider plugin_v1.Provider
-		var value []byte
+		credentialsByProvider[credential.From] = append(
+			credentialsByProvider[credential.From],
+			credential,
+		)
+	}
 
-		if provider, err = resolver.Provider(credential.From); err != nil {
-			resolver.LogFatalf("ERROR: Provider '%s' could not be used! %v", credential.From, err)
+	// Resolve credentials by provider
+	for providerID, credentialsForProvider := range credentialsByProvider {
+		provider, err := resolver.Provider(providerID)
+		if err != nil {
+			resolver.LogFatalf("ERROR: Provider '%s' could not be used! %v", providerID, err)
 		}
 
-		// This provider cannot resolve the named credential
-		if value, err = provider.GetValue(credential.Get); err != nil {
-			errInfo := fmt.Sprintf("ERROR: Resolving credential '%s' from provider '%s' failed: %v",
-				credential.Get,
-				credential.From,
-				err)
+		// Create secretIds slice and credentialBySecretId map
+		var secretIds = make([]string, len(credentialsForProvider))
+		for idx, cred := range credentialsForProvider {
+			secretIds[idx] = cred.Get
+		}
+
+		// Resolves all credentials for current provider
+		providerResponses, err := provider.GetValues(secretIds...)
+		if err != nil {
+			errInfo := fmt.Sprintf(
+				"ERROR: Resolving credentials from provider '%s' failed: %v",
+				provider.GetName(),
+				err,
+			)
 			log.Println(errInfo)
 
 			errorStrings = append(errorStrings, errInfo)
 			continue
 		}
 
-		result[credential.Name] = value
+		// Collect errors from provider responses
+		var hasErrors bool
+		for _, providerResponse := range providerResponses {
+			if providerResponse.Error != nil {
+				hasErrors = true
+				errorStrings = append(errorStrings, providerResponse.Error.Error())
+				continue
+			}
+		}
+		if hasErrors {
+			continue
+		}
+
+		// Set provider responses on result map before returning
+		for _, credential := range credentialsForProvider {
+			credentialName := credential.Name
+			secretValue := providerResponses[credential.Get].Value
+
+			result[credentialName] = secretValue
 
-		if resolver.EventNotifier != nil {
-			resolver.EventNotifier.ResolveCredential(provider, credential.Name, value)
+			if resolver.EventNotifier != nil {
+				resolver.EventNotifier.ResolveCredential(
+					provider,
+					credentialName,
+					secretValue,
+				)
+			}
```

Comment:
I think we need a few more tests for this since we introduced quite a few more logic bifurcation paths